Mathlib/CategoryTheory/Monoidal/Braided.lean

```diff
@@ -123,6 +129,38 @@ theorem braiding_naturality {X X' Y Y' : C} (f : X ⟶ Y) (g : X' ⟶ Y') :
   rw [tensorHom_def' f g, tensorHom_def g f]
   simp_rw [Category.assoc, braiding_naturality_left, braiding_naturality_right_assoc]
 
+theorem yang_baxter (X Y Z : C) :
+    (α_ X Y Z).inv ≫ ((β_ X Y).hom ▷ Z) ≫ (α_ Y X Z).hom ≫
+    (Y ◁ (β_ X Z).hom) ≫ (α_ Y Z X).inv ≫
+    ((β_ Y Z).hom ▷ X) ≫ (α_ Z Y X).hom
+    = (X ◁ (β_ Y Z).hom) ≫ (α_ X Z Y).inv ≫
+      ((β_ X Z).hom ▷ Y) ≫ (α_ Z X Y).hom ≫
+      (Z ◁ (β_ X Y).hom) := by
+  have := (braiding_naturality_right_assoc X (β_ Y Z).hom (α_ Z Y X).hom).symm
+  refine Eq.trans (Eq.symm ?_) (Eq.trans this ?_)
+  · refine Eq.trans (congrArg (. ≫ _) (braiding_tensor_right X Y Z)) ?_
+    simp only [assoc]
+  · refine congrArg (_ ≫ .) ((Iso.eq_comp_inv _).mp ?_)
+    refine Eq.trans (braiding_tensor_right X Z Y) ?_
+    simp only [assoc]
```

Comment:
You should add the attribute `@[simp, reassoc]` to the four lemmas `braiding_tensor_left` et al, so that you could simplify this proof:
```suggestion
  rw [← braiding_tensor_right_assoc X Y Z, ← cancel_mono (α_ Z Y X).inv, assoc, assoc,
    assoc, assoc, assoc, assoc, Iso.hom_inv_id, comp_id,
    ← braiding_naturality_right X (β_ Y Z).hom, braiding_tensor_right]
```
(Note that `@[reassoc (attr := simp)]` would be the correct spelling if the LHS of the lemmas was a composition, but when only the RHS is `@[simp, reassoc]` is useful.)

Reply:
I would rather not touch those at the moment, since this will already need to be fixed up once #6307 is merged

Reply:
I am only suggesting to take the opportunity to fix 4 missing `reassoc` attributes and cleaning up a proof (which would become a sequence of `rw`, which is the easiest to maintain). I do not see any drawback to this.

Reply:
Fair enough